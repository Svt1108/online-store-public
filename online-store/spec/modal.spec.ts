import ModalWindow from "../src/components/modal";

describe("Method showCart", () => {
  const testModalWindow = new ModalWindow();

  test("should be defined", () => {
    expect(testModalWindow.createModal).toBeDefined;
  });

  it("create modal window correctly", () => {
    const data = {
      id: "1",
      name: "Lovely Mexico",
      img: "plate-1.jpg",
      type: "stoneware",
      author: "Grazielli David",
      year: "2022",
      color: "chocolate",
      secondColor: "green",
      quantity: "3",
      popularity: "yes",
      price: "10",
    };
    testModalWindow.createModal(data);
    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
