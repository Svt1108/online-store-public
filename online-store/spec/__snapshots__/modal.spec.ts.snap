// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Method showCart create modal window correctly 1`] = `
"<div class=\\"modal-wrap\\"><div class=\\"modal-window\\"><div class=\\"modal-window__image\\" style=\\"background-image: url(./assets/images/plate-1.jpg);\\"></div><div class=\\"modal-window__text\\">	
    <p class=\\"gray-text text-darken-4 modal-window__title\\">Lovely Mexico</p>	
      <ul class=\\"modal-window__list\\">	
      <li><strong>Author: </strong>Grazielli David</li>	
      <li><strong>Year: </strong>2022</li>	
      <br>	
      <li><strong>Material: </strong>stoneware</li>	
      <li><strong>Color: </strong>chocolate</li>	
      <li><strong>Popularity: </strong>yes</li>	
      <li><strong>Quantity: </strong>3</li>	
      </ul>	
      <p class=\\"grey-text text-darken-4 modal-window__title\\">10 â‚¬</p></div></div></div>"
`;
